sklearn/model_selection/_search.py

```diff
@@ -793,14 +794,17 @@ def evaluate_candidates(candidate_params, cv=None, more_results=None):
                         "splits, got {}".format(n_splits, len(out) // n_candidates)
                     )
 
+                all_candidate_params.extend(candidate_params)
+                all_out.extend(out)
+                _warn_about_fit_failures(all_out, self.error_score)
```

Comment:
Multiple calls to `evaluate_candidates` in sucessive halving will lead to multiple warnings that can reference the same warnings. I think a good sort term solution is to call `_warn_about_fit_failures(out, self.error_score)`.

Reply:
What we discuss with @lesteve IRL is that some of the calls could give rise to some `nan` and some others will not. This might not be that bad to raise a warning at each evaluation in this case?

Reply:
> I think a good sort term solution is to call _warn_about_fit_failures(out, self.error_score).

So I did this in my last commit